map/traffic_manager.hpp

```diff
@@ -80,13 +81,33 @@ class TrafficManager final
   void OnMwmDelete(MwmSet::MwmId const & mwmId);
 
 private:
+  struct CacheEntry
+  {
+    CacheEntry();
+    CacheEntry(time_point<steady_clock> const & requestTime);
+
+    bool m_isLoaded;
+    size_t m_dataSize;
+
+    time_point<steady_clock> m_lastActiveTime;
+    time_point<steady_clock> m_lastRequestTime;
+    time_point<steady_clock> m_lastResponseTime;
+
+    int m_retriesCount;
+    bool m_isWaitingForResponse;
+
+    traffic::TrafficInfo::Availability m_lastAvailability;
+  };
+
   void ThreadRoutine();
   bool WaitForRequest(vector<MwmSet::MwmId> & mwms);
 
   void OnTrafficDataResponse(traffic::TrafficInfo && info);
   void OnTrafficRequestFailed(traffic::TrafficInfo && info);
 
-private:
+  void UpdateActiveMwms(m2::RectD const & rect, vector<MwmSet::MwmId> & lastMwmsByRect,
```

Comment:
Please, add a comment that activeMwms may be only m_activeRoutingMwms or m_activeDrapeMwms, and that mutex is locked inside the function, therefore it should be called without mutex.